### PR TITLE
[Birthday] Migrate to v3

### LINKED ISSUE #248: [Birthday] Migrate to v3
As per title.

- [x] Create `birthday` folder in `cogs/` folder, and pre-populate with v2 `birthday.py`.
- [x] Move setup code into `__init__.py` within the same folder.
- [x] Refactor to use Config v3 (see Red-DiscordBot documentation for examples, use 5842647 as identifier).
- [x] Refactor all methods to use new discord.py APIs.
- [ ] Do manual testing.
- [x] Create v2 to v3 converter, or look into the dataconverter cog.

### PR DESCRIPTION
**Pull request by Injabie3**
### Type
- [x] Migration

### Description of the changes
Resolves #248.  See this [diff](https://github.com/SFUAnime/Ren/pull/252/files/540fcba94f6a0ee4a3d7b4dcab0118d4e6c63c50..HEAD) for v2 to v3 refactoring.

### CODE REVIEW

**Comment by Injabie3** [comment]
Will be testing this throughout the next few days, put some fake birthdays in to see if the bot will add and remove correctly.

**Comment by Injabie3** [comment]
It's been a few days, and roles are being removed and added as expected, even across bot restarts.

File: cogs/birthday/birthday.py [hunk 1]
```diff
@@ -0,0 +1,459 @@
...
+                "Could not add %s#%s (%s) to birthday role, does the bot "
+                "have enough permissions?",
+                member.name,
+                member.discriminator,
+                member.id,
+                exc_info=True)
+            await ctx.send(
+                ":negative_squared_cross_mark: **Birthday - Add**: Could "
+                "not add **{}** to the list, the bot does not have enough "
+                "permissions to do so!".format(member.name))
```
Comment (KaguneAstra) [inline cogs/birthday/birthday.py]: Nit: specify which permission is needed

File: cogs/birthday/birthday.py [hunk 2]
```diff
@@ -0,0 +1,459 @@
...
+                "Could not remove %s#%s (%s) from the birthday role, does "
+                "the bot have enough permissions?",
+                member.name,
+                member.discriminator,
+                member.id,
+                exc_info=True)
+            await ctx.send(
+                ":negative_squared_cross_mark: **Birthday - Delete**: "
+                "Could not remove **{}** from the role, the bot does not "
+                "have enough permissions to do so!".format(member.name))
```
Comment (KaguneAstra) [inline cogs/birthday/birthday.py]: Same nit as above; specify manage roles permission

File: cogs/birthday/birthday.py [hunk 3]
```diff
@@ -0,0 +1,459 @@
...
+                            # another cog saves roles.
+                            continue
+
+                        # Update the list.
+                        await self.config.member(member).isAssigned.set(False)
+
+    ##################################################################
+    # Event Loop - Check to see if we need to add people to the role #
+    ##################################################################
+    async def _dailyAdd(self):  # pylint: disable=too-many-branches
```
Comment (KaguneAstra) [inline cogs/birthday/birthday.py:402]: Function is missing Doc string

File: cogs/birthday/birthday.py [hunk 4]
```diff
@@ -0,0 +1,459 @@
...
+
+        # Avoid having data modified by other methods.
+        # When we acquire the lock for all members, it also prevents lock for guild
+        # from being acquired, which is what we want.
+        membersLock = self.config.get_members_lock()
+
+        async with membersLock:
+            # Check each guild.
+            for guild in guilds:
+                # Make sure the guild is configured with birthdya role.
```
Comment (KaguneAstra) [inline cogs/birthday/birthday.py]: **birthday

**Review by KaguneAstra — APPROVED** [review]
Some nits here and there but LGTM

**Comment by Injabie3** [comment]
Addressed aforementioned feedback, merging

**Pull request merged**